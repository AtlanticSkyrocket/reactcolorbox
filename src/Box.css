.Box {
  display: flex;
  align-items: center; /* Vertically center the items */
  justify-content: center; /* Horizontally center the items */
}
.Box-color {
  margin-right: 10px;
}

.BoxRemove-btn {
  margin-left: 10px;
}