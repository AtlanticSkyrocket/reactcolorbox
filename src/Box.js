import React from "react";
import './Box.css';

function Box({removeBox, backgroundColor, width=100, height=100, id}){
  return (
    <div className="Box">
          <div className="Box-color" key={id} id={id} style={{backgroundColor: backgroundColor, width: `${width}px`, height:`${height}px`}}/>
          <button className="BoxRemove-btn"  onClick={() => removeBox(id)}>X</button>
    </div>

  )
}

export default Box;