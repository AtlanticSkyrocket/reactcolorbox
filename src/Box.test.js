import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Box from './Box';



const mockRemoveBox = jest.fn();

it('renders without crashing', () => {
  render(<Box removeBox={mockRemoveBox} backgroundColor={"#000000"} width={100} height={100} id="1" />);
});


it('matches snapshot', () => {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});

it('renders with custom width and height', () => {
  render(<Box id={2} removeBox={mockRemoveBox} backgroundColor="blue" width={150} height={150} />);
  
  const boxElement = screen.getByText('X');
  console.log(boxElement);
  expect(boxElement).toBeInTheDocument();
});

it('calls removeBox when the remove button is clicked', () => {
  render(<Box removeBox={mockRemoveBox} backgroundColor={"#000000"} width={100} height={100} id="1" />);

  const removeBtn = screen.getByText('X');

  fireEvent.click(removeBtn);

  expect(mockRemoveBox).toBeCalledWith('1');
});