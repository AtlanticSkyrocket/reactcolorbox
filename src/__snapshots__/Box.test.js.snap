// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      class="Box-color"
      style="width: 100px; height: 100px;"
    />
    <button
      class="BoxRemove-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
